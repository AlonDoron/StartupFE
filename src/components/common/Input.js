import React from "react";
import { View, StyleSheet } from "react-native";
import { TextInput, Text } from "react-native-paper";

const Input = (props) => {
  return (
    <View style={styles.textInput}>
      <TextInput
        label={props.label}
        mode="outlined"
        onChange={props.onChange}
      />
      <Text>{props.errors}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  textInput: {
    width: "80%",
    alignSelf: "center",
    paddingTop: "5%",
  },
});

export default Input;
