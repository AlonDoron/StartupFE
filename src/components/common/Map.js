import React from "react";
import { StyleSheet } from "react-native";
import MapView from "react-native-maps";

const Map = ({ currentLocation }) => {
  return (
    <MapView
      style={styles.map}
      initialRegion={{
        ...currentLocation,
        latitudeDelta: 0.01,
        longitudeDelta: 0.01,
      }}
    />
  );
};

const styles = StyleSheet.create({
  map: {
    height: "90%",
  },
});

export default Map;
