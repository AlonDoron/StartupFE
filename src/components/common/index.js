export { default as Input } from "./Input";
export { default as LoadingIndicator } from "./LoadingIndicator";
export { default as Map } from "./Map";
