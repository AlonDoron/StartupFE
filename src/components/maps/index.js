export { default as MapviewWrapper } from "./MapviewWrapper";
