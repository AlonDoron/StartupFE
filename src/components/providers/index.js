export { default as ProvidersList } from "./ProvidersList";
