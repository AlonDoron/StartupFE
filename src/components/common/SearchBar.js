import React, { useState } from "react";
import { useForm } from "../../hooks";
import { StyleSheet, View, Modal, Text } from "react-native";
import { Button } from "react-native-paper";
import Input from "./Input";

import { Entypo } from "@expo/vector-icons";

const SearchBar = (props) => {
  const [modalVisible, setModalVisible] = useState(false);
  const [values, handleChange, handleSubmit] = useForm();

  const message = (radius, job) => {
    console.log(`you searcing for ${job} within radius of ${radius}`);
  };

  return (
    <View style={styles.barContainer}>
      <View style={{ justifyContent: "center", alignItems: "center" }}>
        <Entypo
          name="magnifying-glass"
          size={24}
          style={styles.icon}
          onPress={() => setModalVisible(true)}
        />
      </View>
      <Modal visible={modalVisible} animationType={"slide"}>
        <View
          style={{
            justifyContent: "center",
            alignItems: "center",
            marginVertical: 30,
          }}
        >
          <Entypo
            name="back"
            size={24}
            style={styles.icon}
            onPress={() => setModalVisible(false)}
          />

          <Input
            name="byRadius"
            value={values.byRadius || ""}
            onChange={(name, value) => handleChange(name, value)}
            label="Search By Radius"
            keyboardType="number-pad"
          />

          <Input
            name="bySubject"
            value={values.bySubject || ""}
            onChange={(name, value) => handleChange(name, value)}
            label="Search By Subject"
          />
          <Button
            title="Search"
            onPress={() => message(values.byRadius, values.bySubject)}
          >
            Search
          </Button>
        </View>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  barContainer: {
    width: "100%",
    height: "10%",
    flexDirection: "row",
    justifyContent: "space-evenly",
  },
  con: {
    flex: 1,
  },
});

export default SearchBar;
