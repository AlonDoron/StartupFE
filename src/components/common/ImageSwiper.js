import React, { useState } from "react";
import { Layout, Text } from "@ui-kitten/components";
import {
  StyleSheet,
  ScrollView,
  Image,
  useWindowDimensions,
} from "react-native";

const ImageSwiper = ({ images }) => {
  const window = useWindowDimensions();
  const [isImgActive, setImgActive] = useState(0);

  const handleImageScroll = ({ nativeEvent }) => {
    const slide = Math.ceil(
      nativeEvent.contentOffset.x / nativeEvent.layoutMeasurement.width
    );
    if (slide !== isImgActive) setImgActive(slide);
  };

  return (
    <Layout
      style={{
        flex: 1,
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 5,
        },
        shadowOpacity: 0.36,
        shadowRadius: 6.68,

        elevation: 11,
      }}
    >
      <ScrollView
        style={{
          width: window.width,
          height: window.width * 0.6,
        }}
        pagingEnabled
        horizontal
        showsHorizontalScrollIndicator={false}
        onScroll={handleImageScroll}
      >
        {images.map((image, index) => (
          <Image
            key={index}
            style={{
              width: window.width,
              height: "100%",
              resizeMode: "cover",
            }}
            source={image}
          />
        ))}
      </ScrollView>
      <Layout
        style={{
          flexDirection: "row",
          position: "absolute",
          bottom: 0,
          alignSelf: "center",
          backgroundColor: "transparent",
        }}
      >
        {images.map((item, index) => (
          <Text
            key={index}
            style={[
              { fontSize: window.width / 30, margin: 3 },
              index == isImgActive ? { color: "#000" } : { color: "#ff0000" },
            ]}
          >
            &#11044;
          </Text>
        ))}
      </Layout>
    </Layout>
  );
};

const styles = StyleSheet.create({
  listContaier: {
    //  width:window.wi
  },
});

export default ImageSwiper;
