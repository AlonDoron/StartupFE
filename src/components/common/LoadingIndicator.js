import React from "react";
import { StyleSheet, View } from "react-native";
import { Spinner } from "@ui-kitten/components";

const LoadingIndicator = (props) => {
  return (
    <View style={[props.style, styles.indicator]}>
      <Spinner size="medium" />
    </View>
  );
};

const styles = StyleSheet.create({
  indicator: {
    justifyContent: "center",
    alignItems: "center",
  },
});

export default LoadingIndicator;
