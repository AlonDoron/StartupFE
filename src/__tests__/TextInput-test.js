import React from "react";
import { Input } from "../components/common";
import { render, fireEvent } from "@testing-library/react-native";

import renderer from "react-test-renderer";

const setup = () => {
  const utils = render(<Input />);
  const input = utils.getByTestId("text-input");
  return {
    input,
    ...utils,
  };
};

describe("It should display values", () => {
  it("Should display empty value", () => {
    const { input } = setup();

    fireEvent.changeText(input, "");
    expect(input.props.value).toBe("");
  });

  it("Should display `Cherry` value", () => {
    const { input } = setup();
    const string = "Cherry";

    fireEvent.changeText(input, string);
    expect(input.props.value).toBe(string);
  });
});

describe("Render Input", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
