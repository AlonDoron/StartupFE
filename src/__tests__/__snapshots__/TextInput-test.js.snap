// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "paddingTop": "5%",
      "width": "80%",
    }
  }
>
  <View
    style={Object {}}
  >
    <View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 6,
            },
            Object {
              "backgroundColor": "#f6f6f6",
              "borderColor": "rgba(0, 0, 0, 0.54)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "minHeight": 64,
            "paddingBottom": 0,
            "paddingTop": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "textAlign": "left",
                "zIndex": 1,
              },
              Object {
                "height": 56,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlignVertical": "center",
              },
            ]
          }
          testID="text-input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        undefined,
      ]
    }
  />
</View>
`;
